### DISCUSSION TITLE
How to migrate pocketbase in fly.io in production.

### DISCUSSION THREAD

**Discussion by xrgouda** [pocketbase/pocketbase#1532] [Q&A] [answered]
**How I can migrate to latest pocketbase version (v0.11.0) from (v0.10.4), without losing my database ?**

And how I can make backup for the database on fly.io and download it or restore it again ?

Finally how I can do that without having go environment or go project in my Mac, it's only pocketbase instance running on fly.io server ?

**Comment by ganigeorgiev** [comment, marked as answer]
1. Make a backup/download your remote `pb_data` to your device:
    ```sh
    # this will register a ssh key with your local agent (if you haven't already)
    flyctl ssh issue --agent

    # proxies connections to a fly VM through a Wireguard tunnel
    flyctl proxy 10022:22

    # run in a separate terminal to copy the pb_data directory from the remote
    # (to restore from local -> remote you can reverse the arguments)
    scp -r -P 10022 root@localhost:/pb/pb_data  /your/local/path/where/to/copy/pb_data
    ```

2. Download locally the [latest executable for your platform](https://github.com/pocketbase/pocketbase/releases/tag/v0.11.0) from the "Assets" section and test the changes with the new version. To start PocketBase with specific pb_data you can use the `--dir` flag:
   ```
   ./pocketbase serve --dir="/your/local/path/to/pb_data"
   ```

3. If everything is OK, you can now update your Dockerfile OR change the env variable under `build.args` in your `fly.toml` to the `"0.11.0"` version and then just run `flyctl deploy` to redploy your updated configuration.

For performing others and more advanced tasks, you can also ssh in your vm with `fly ssh console`.

**Reply by xrgouda** [reply]
Thank you for replying,, But when I wrote this in terminal

> flyctl proxy 10022:22

I got this error :

 Error .internal: host was not found in DNS

**Reply by ganigeorgiev** [reply]
I'm not sure what could be the reason for the error. Maybe there is a firewall or something else Mac specific? Did you run `flyctl ssh issue --agent` before that?

**Reply by ganigeorgiev** [reply]
Also note that the flyctl commands need to be executed in the directory where your fly.toml is located (_there are probably flags that you could use to specify the config; you can try running `flyctl --help` to check all options/commands_).

**Reply by ganigeorgiev** [reply]
Alternatively, you can try downloading your pb_data with the sftp client as mentioned in https://github.com/pocketbase/pocketbase/discussions/537#discussioncomment-3914729.

**Reply by xrgouda** [reply]
Thank you, It's working now.. But why I am getting empty folder with no data when I do these commands.

Like i am getting **lost+found** folder as backup for my data but it's empty.

**Reply by ganigeorgiev** [reply]
I'm confused. What command did you end up using, the scp one or the sftp? Could you provide the steps/commands (and their output) you've executed so far?

**Reply by xrgouda** [reply]
1 - flyctl ssh issue --agent

Choose my organization & Entering my email

2- flyctl proxy 10022:22

Open another terminal

3- scp -r -P 10022 root@localhost:/pb/pb_data  /mydirectory/to/put/backup/pb_data

But result here is pb_data folder and inside **lost+found** folder and it's empty..

This is about backup the data.

About deploying I edit fly.toml file and update pocketbase to 0.11.0

Then -> flyctl deploy

and it should be deploying but it's giving me that in log almost for 30 minute and did't deploy yet

**proxy[31dc173f] ams [warn] Instance reached connections hard limit of 25**

is that normal.. or I did something wrong ?

**Reply by ganigeorgiev** [reply]
I'm not able to reproduce it.

I've just deployed a brand new vm following the instructions from https://github.com/pocketbase/pocketbase/discussions/537.

I've even updated my `flyctl` to:
```
flyctl v0.0.443 linux/amd64 Commit: 63f03d09 BuildDate: 2023-01-05T21:09:40Z
```

The output of the scp command should be something like:

```sh
data.db-shm              100%   32KB 192.6KB/s   00:00
logs.db-shm              100%   32KB 387.5KB/s   00:00
logs.db                  100% 4096    53.9KB/s   00:00
data.db                  100% 4096    54.3KB/s   00:00
data.db-wal              100%  189KB 477.1KB/s   00:00
logs.db-wal              100%  334KB 619.1KB/s   00:00
```

Either there is something wrong with your fly.toml/Dockerfile or it could be some obscure Mac issue, I'm not really sure (_I'm testing on Fedora 35_).

**Reply by ganigeorgiev** [reply]
You can also try to inspect your logs from the fly.io "Dashboard > (app) > Monitoring" and check what is the state of your running vm.

**Reply by ganigeorgiev** [reply]
_Just for info, if you haven't found the source of the problem yet, you can change the default hard connections limit by updating your fly.toml `services.services.concurrency.hard_limit`._

**Reply by xrgouda** [reply]
Thank you,, I just upload the new version in fly.io..

But unfortunately my old documents and data is gone,, any way that I get them back ?

**Reply by ganigeorgiev** [reply]
If you haven't deleted your old app, you should be able to restore its volume snapshot from a previous state (they are kept for 5 days) - https://fly.io/docs/reference/volumes/#snapshots-and-restores.
